tests/unit/s2n_security_policies_test.c

```diff
@@ -83,29 +83,38 @@ int main(int argc, char **argv)
                     struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER);
                     EXPECT_NOT_NULL(client_conn);
                     EXPECT_NOT_NULL(server_conn);
-                    EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, security_policy_selection[policy_index].version));
-                    EXPECT_SUCCESS(s2n_config_set_verification_ca_location(config, S2N_DEFAULT_TEST_CERT_CHAIN, NULL));
-                    EXPECT_NOT_NULL(config->default_certs_by_type.certs[S2N_PKEY_TYPE_RSA]);
-                    EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
-                    EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
-
-                    client_conn->actual_protocol_version = S2N_TLS13;
-                    server_conn->actual_protocol_version = S2N_TLS13;
-                    client_conn->secure.cipher_suite = &tls_13_ciphers[i];
-                    server_conn->secure.cipher_suite = &tls_13_ciphers[i];
-
-                    struct s2n_signature_scheme chosen_scheme = {0};
-
-                    if (s2n_is_rsa_pss_signing_supported()) {
-                        /* If RSA PSS signing is supported, then we should always be able to select a default Signature
-                         * Scheme for RSA Certs for TLS 1.3 */
-                        EXPECT_SUCCESS(s2n_tls13_default_sig_scheme(client_conn, &chosen_scheme));
-                        EXPECT_SUCCESS(s2n_tls13_default_sig_scheme(server_conn, &chosen_scheme));
+
+                    if (security_policy_selection[policy_index].security_policy->minimum_protocol_version == S2N_TLS13
+                            && !s2n_is_tls13_fully_supported()) {
+                        /* We purposefully do not allow users to configure Security Policies with a minimum allowed TLS
+                         * Version of TLS 1.3, if TLS 1.3 algorithms aren't fully supported by the libcrypto we're using */
+                        EXPECT_FAILURE(s2n_config_set_cipher_preferences(config, security_policy_selection[policy_index].version));
```

Comment:
Can we test this failure behavior once, and then restrict the rest of these tests to only TLS1.3? Mixing this failure case in with other tests makes them harder to follow.

Reply:
For this specific test, we're iterating through all Security Policies and checking their properties, and since this test is reproducing the RSA PSS issue when s2n is compiled against Openssl 1.0.2 that the CRT ran into, I'd prefer not to skip this test. I've refactored it slightly to make the condition more generic and hopefully easier to follow.